src/dmd/backend/machobj.d

```diff
@@ -1910,6 +1925,26 @@ int Obj_getsegment(const(char)* sectname, const(char)* segname,
     return seg;
 }
 
+/********************************
+ * Memoize seg index.
+ * Params:
+ *      seg = value to memoize if it is not already set
+ *      sectname = section name
+ *      segname = segment name
+ *      align_ = section alignment
+ *      flags = S_????
+ * Returns:
+ *      seg index
+ */
+private extern (D)
+int getsegment2(ref int seg, const(char)* sectname, const(char)* segname,
+        int align_, int flags)
+{
+    if (seg == UNKNOWN)
+        seg = Obj_getsegment(sectname, segname, align_, flags);
```

Comment:
Wait, is `Obj_getsegment` only used from within this function? If so, can you make it static inside this one?

Reply:
It's also used by dwarfdbginf.d